base/char.jl

```diff
@@ -202,10 +204,7 @@ typemin(::Type{Char}) = bitcast(Char, typemin(UInt32))
 
 size(c::AbstractChar) = ()
 size(c::AbstractChar, d::Integer) = d < 1 ? throw(BoundsError()) : 1
-ndims(c::AbstractChar) = 0
-ndims(::Type{<:AbstractChar}) = 0
 length(c::AbstractChar) = 1
-IteratorSize(::Type{Char}) = HasShape{0}()
```

Comment:
NB: here `IteratorSize` was defined for `Char`. With this PR it's defined for all `AbstractChar` subtypes (making this kinda a minor change). However that's OK, because the this method was supposed to be `IteratorSize(::Type{AbstractChar}) = HasShape{0}()` in the first place. Refer to:
* PR #29819
* PR #40877
* PR #41073

Reply:
`ndims` being defined for `AbstractChar` implies `IteratorSize` should be treated the same, I'd say.